src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs

```diff
@@ -118,35 +113,29 @@ public override string FullPath
             }
         }
 
-        private static string GetAssemblyNameFromPath(string path)
+        public override string Display
         {
-            // AssemblyName.GetAssemblyName(path) is not available on CoreCLR.
-            // Use our metadata reader to do the equivalent thing.
-            using (var reader = new PEReader(FileUtilities.OpenRead(path)))
+            get
             {
-                var metadataReader = reader.GetMetadataReader();
-                var assemblyDefinition = metadataReader.GetAssemblyDefinition();
-                return metadataReader.GetString(assemblyDefinition.Name);
+                if (_lazyDisplayName == null)
+                {
+                    _lazyDisplayName = Path.GetFileName(this.FullPath);
+                }
+
+                return _lazyDisplayName;
             }
         }
 
-        public override string Display
+        public override string Id
```

Comment:
@tmeschter same comment as before. can this return object rather than string? Id shouldn't have any meaning